framework/include/utils/MathFVUtils.h

```diff
@@ -430,10 +430,16 @@ template <typename Scalar, typename Vector>
 Scalar
 rF(const Scalar & phiC, const Scalar & phiD, const Vector & gradC, const RealVectorValue & dCD)
 {
-  if ((phiD - phiC) == 0)
-    return 1e6 * MathUtils::sign(gradC * dCD) + 0 * (gradC * dCD + phiD + phiC);
+  const auto fgrad = phiD - phiC;
+  const auto fgradC = gradC * dCD;
+  constexpr Real coeff = 1e3;
 
-  return 2. * gradC * dCD / (phiD - phiC) - 1.;
+  if (std::abs(fgradC) >= coeff * std::abs(fgrad))
+    // Second term (multiplied by zero) is to keep same sparsity pattern as else branch
+    return 2 * coeff * MathUtils::sign(fgradC) * MathUtils::sign(fgrad) +
+           0 * (gradC * dCD + phiD + phiC);
```

Comment:
Why do we need this term?

Reply:
says in the comment haha

Reply:
Ahhh, damn Peter: learn to read!
OF had a similar issue at some point, I had to add a 0.0*laplacian(variable) to make sure the offdiagonals were initialized.

Reply:
Shouldn't vec(0)*gradC be enough? It already touches cell C and D, so we could save some operations here. Just a suggestion.

Reply:
:+1: changed

Reply:
Still cant change this to resolved, so feel free to resolve this of just merge if needed.